crates/bevy_math/src/common_traits.rs

```diff
@@ -71,6 +71,85 @@ impl VectorSpace for f32 {
     const ZERO: Self = 0.0;
 }
 
+/// A type consisting of formal sums of elements from `V` and `W`.
+///
+/// That is, values `Sum(v, w)` should be thought of as `v + w`.
```

Comment:
I think more docs might be useful to clue people into the math.
Ex: A formal sum does not need to have a defined resulting value of the addition, it's just the addition itself.
Or that doing the sum of two vector spaces gives a vector space with the sum of the dimensions.